src/neo/SmartContract/Callbacks/MethodCallback.cs

```diff
@@ -0,0 +1,35 @@
+using Neo.IO;
+using Neo.Ledger;
+using Neo.SmartContract.Manifest;
+using System;
+using System.Linq;
+using Array = Neo.VM.Types.Array;
+
+namespace Neo.SmartContract.Callbacks
+{
+    internal class MethodCallback : SyscallCallback
+    {
+        private readonly ContractState contract;
+        private readonly ContractMethodDescriptor method;
+
+        public override int ParametersCount => method.Parameters.Length;
+
+        public MethodCallback(ApplicationEngine engine, UInt160 hash, string method)
+            : base(ApplicationEngine.System_Contract_Call)
+        {
+            if (method.StartsWith('_')) throw new ArgumentException();
```

Comment:
Maybe we should prevent only `_initialize` ?

Reply:
No one will name a public method starting with `_`. So let's reserve all of them.